### PR TITLE
Fix for #107

### PR DESCRIPTION
**Pull request by ostracon**
Based on feedback/hints in #107 and some head-bashing, generated this PR, with which the full functionality of the Daikin AC units via Niko is avaiable in HA.

### CODE REVIEW

File: custom_components/nhc2/entities/generic_hvac_climate.py [hunk 1]
```diff
@@ -147,13 +176,31 @@ async def async_set_temperature(self, **kwargs):
...
         if hvac_mode == HVACMode.AUTO or hvac_mode == HVACMode.HEAT_COOL:
             self._device.set_operation_mode(self._gateway, PROPERTY_OPERATION_MODE_VALUE_AUTO)
+        if hvac_mode == HVACMode.FAN_ONLY:
+            self._device.set_operation_mode(self._gateway, PROPERTY_OPERATION_MODE_VALUE_FAN)
+        if hvac_mode == HVACMode.DRY:
+            self._device.set_operation_mode(self._gateway, PROPERTY_OPERATION_MODE_VALUE_DRY)
+
+        self.on_change()
+
+    async def async_turn_on(self) -> None:
```
Comment (tijsverkoyen) [inline custom_components/nhc2/entities/generic_hvac_climate.py:198]: I don't think these methods are supported by Home Assistant. As I can't find them in the documentation for the Climate Entity: https://developers.home-assistant.io/docs/core/entity/climate/#methods

File: .gitignore [hunk 2]
```diff
@@ -14,4 +14,5 @@ home-assistant.log
 scenes.yaml
 scripts.yaml
 secrets.yaml
-debugging
\ No newline at end of file
+debugging
+.idea/
```
[the remark this answers is not in the document]
Comment (ostracon) [reply .gitignore]: Ah, my bad - reverted this change

**Comment by ostracon** [comment]
Just to note on the `.exception` vs `.error` - it seems the error I previously reported is related to the energy component; disabling the device as recommended in the README removes the error, but for reference, with `.exception`, a fuller trace is:

```
2024-01-11 09:37:19.380 ERROR (Thread-5 (_thread_main)) [custom_components.nhc2.nhccoco.coco] 'NoneType' object has no attribute 'loop'
Traceback (most recent call last):
  File "/config/custom_components/nhc2/nhccoco/coco.py", line 150, in _on_message
    self._device_instances[device[MQTT_DATA_PARAMS_DEVICES_UUID]].on_change(topic, device)
  File "/config/custom_components/nhc2/nhccoco/devices/generic_energyhome.py", line 76, in on_change
    callback()
  File "/config/custom_components/nhc2/entities/generic_energyhome_electrical_power_to_grid.py", line 53, in on_change
    self.schedule_update_ha_state()
  File "/usr/src/homeassistant/homeassistant/helpers/entity.py", line 1157, in schedule_update_ha_state
    self.hass.loop.call_soon_threadsafe(self.async_write_ha_state)
    ^^^^^^^^^^^^^^
AttributeError: 'NoneType' object has no attribute 'loop'`
```

File: custom_components/nhc2/nhccoco/devices/device.py [hunk 3]
```diff
@@ -1,3 +1,5 @@
+from typing import Union, List
```
Comment (tijsverkoyen) [inline custom_components/nhc2/nhccoco/devices/device.py:1]: Why this `List` and not `list`. 
I'm not a python expert, and throughout the codebase `list` is used. Can you change this for the sake of having consitency

File: custom_components/nhc2/nhccoco/devices/device.py [hunk 4]
```diff
@@ -123,7 +125,7 @@ def extract_property_definition(self, property_key: str) -> str:
                 return property_definition_object[property_key]
         return None
 
-    def extract_property_definition_description_choices(self, property_key: str) -> str:
+    def extract_property_definition_description_choices(self, property_key: str) -> Union[List, None]:
```
Comment (tijsverkoyen) [inline custom_components/nhc2/nhccoco/devices/device.py:128]: Why this `List` and not `list`. 
I'm not a python expert, and throughout the codebase `list` is used. Can you change this for the sake of having consitency

File: custom_components/nhc2/nhccoco/devices/generic_hvac.py [hunk 5]
```diff
@@ -1,3 +1,5 @@
+from typing import List
```
Comment (tijsverkoyen) [inline custom_components/nhc2/nhccoco/devices/generic_hvac.py:1]: Why this `List` and not `list`. 
I'm not a python expert, and throughout the codebase `list` is used. Can you change this for the sake of having consitency

File: custom_components/nhc2/nhccoco/devices/generic_hvac.py [hunk 6]
```diff
@@ -39,6 +41,10 @@ def supports_program(self) -> bool:
     def operation_mode(self) -> str:
         return self.extract_property_value(PROPERTY_OPERATION_MODE)
 
+    @property
+    def possible_operation_modes(self) -> List[str]:
```
Comment (tijsverkoyen) [inline custom_components/nhc2/nhccoco/devices/generic_hvac.py:45]: Why this `List` and not `list`. 
I'm not a python expert, and throughout the codebase `list` is used. Can you change this for the sake of having consitency

**Comment by ostracon** [comment]
Addressed the outstanding review comments in [7b8c47c](https://github.com/joleys/niko-home-control-II/pull/109/commits/7b8c47c482e39abbcd086b026b4596c6d7111d87)

File: custom_components/nhc2/nhccoco/devices/generic_hvac.py (hunk 5 above)
Comment (tijsverkoyen) [reply custom_components/nhc2/nhccoco/devices/generic_hvac.py:1]: This one is still here.

File: custom_components/nhc2/nhccoco/devices/generic_hvac.py (hunk 6 above)
Comment (tijsverkoyen) [reply custom_components/nhc2/nhccoco/devices/generic_hvac.py:45]: This one is still `List`

**Review by tijsverkoyen — CHANGES_REQUESTED** [review]

**Comment by ostracon** [comment]
Oops, apologies - [f0e0398](https://github.com/joleys/niko-home-control-II/pull/109/commits/f0e03981b5a99e2309d5c0a217a84c2de985aa0b) has those removed

**Comment by ostracon** [comment]
Thanks for your patience!

**Pull request merged**